src/Symfony/Component/Translation/Translator.php

```diff
@@ -150,7 +150,7 @@ public function addResource(string $format, $resource, string $locale, string $d
     public function setLocale(string $locale)
     {
         $this->assertValidLocale($locale);
-        $this->locale = $locale ?? (class_exists(\Locale::class) ? \Locale::getDefault() : 'en');
+        $this->locale = $locale;
```

Comment:
in 4.4, we deprecated calling setLocale with `null`

we could make the empty string fallback to `\Locale::getDefault()`, but there are tests that fail if we do.

Reply:
ref #38230

we could still deprecate empty string on 4.4 isnt it? also the trait currenly stll allows `""` and `"0"`, perhaps we should consider a shared `LocaleAwareTrait`

Reply:
we can't deprecateany more on 4.4
looks like we (I) missed why #38230 was proposed...